server/routes/yourCompletedReports/yourCompletedReports.ts

```diff
@@ -1,56 +1,57 @@
 import { Request, Response } from 'express'
 import mojPaginationFromPageResponse, { pageRequestFrom } from '../../utils/mojPagination/pagination'
-import {
-  ReportedAdjudication,
-  ReportedAdjudicationFilter,
-  ReportedAdjudicationStatus,
-  reportedAdjudicationStatuses,
-} from '../../data/ReportedAdjudicationResult'
+import { ReportedAdjudication, reportedAdjudicationStatuses } from '../../data/ReportedAdjudicationResult'
 import { ApiPageResponse } from '../../data/ApiData'
 import ReportedAdjudicationsService from '../../services/reportedAdjudicationsService'
 import adjudicationUrls from '../../utils/urlGenerator'
 import {
-  reportedAdjudicationFilterFromRequestParameters,
-  uiFilterFromReportedAdjudicationFilter,
+  filterFromUiFilter,
+  UiFilter,
+  uiFilterFromBody,
+  uiFilterFromRequest,
+  validate,
 } from '../../utils/adjudicationFilterHelper'
+import { FormError } from '../../@types/template'
 
 export default class YourCompletedReportsRoutes {
   constructor(private readonly reportedAdjudicationsService: ReportedAdjudicationsService) {}
 
   private renderView = async (
     req: Request,
     res: Response,
-    filter: ReportedAdjudicationFilter,
-    results: ApiPageResponse<ReportedAdjudication>
+    uiFilter: UiFilter,
+    results: ApiPageResponse<ReportedAdjudication>,
+    errors: FormError[]
   ): Promise<void> => {
     return res.render(`pages/yourCompletedReports`, {
       yourCompletedReports: results,
-      filter: uiFilterFromReportedAdjudicationFilter(filter),
+      filter: uiFilter,
       statuses: reportedAdjudicationStatuses,
       pagination: mojPaginationFromPageResponse(
         results,
         new URL(`${req.protocol}://${req.get('host')}${req.originalUrl}`)
       ),
+      errors,
     })
   }
 
   view = async (req: Request, res: Response): Promise<void> => {
-    const filter = reportedAdjudicationFilterFromRequestParameters(req)
+    const uiFilter = uiFilterFromRequest(req)
+    const filter = filterFromUiFilter(uiFilter)
     const results = await this.reportedAdjudicationsService.getYourCompletedAdjudications(
       res.locals.user,
       filter,
       pageRequestFrom(20, +req.query.pageNumber || 1)
     )
-    return this.renderView(req, res, filter, results)
+    return this.renderView(req, res, uiFilter, results, [])
   }
 
   submit = async (req: Request, res: Response): Promise<void> => {
-    return res.redirect(
-      adjudicationUrls.yourCompletedReports.urls.filter(
-        req.body.fromDate.date,
-        req.body.toDate.date,
-        req.body.status as ReportedAdjudicationStatus
-      )
-    )
+    const uiFilter = uiFilterFromBody(req)
```

Comment:
I like this explicit conversion to a type - i think it i much more readable actually!